### ISSUE TITLE
Masking policy: failed to parse result of describe: sql: expected 12 destination arguments in Scan, not 11

### ISSUE THREAD

**Issue by ayoul3** [Snowflake-Labs/terraform-provider-snowflake#2362]
### Terraform CLI and Provider Versions

I tested on 0.68.1 and above

### Terraform Configuration

```terraform
resource "snowflake_table_column_masking_policy_application" "application" {

  table          = "table_test"
  column         = "secret"
  masking_policy = snowflake_masking_policy.policy.qualified_name
}
```

### Expected Behavior

Plan/Apply the masking policy on the column "secret" on the "table_test"

### Actual Behavior

Error: failed to parse result of describe: sql: expected 12 destination arguments in Scan, not 11

### Steps to Reproduce

1. `terraform plan`

### How much impact is this issue causing?

High

### Additional Information

The snowflake_masking_policy works fine. it's the snowflake_table_column_masking_policy_application that seems to be in error

**Comment by sfc-gh-asawicki** [comment]
Hey @ayoul3. Thanks for reporting the issue.

Please provide the full debug logs for the Terraform run (you can use TF_LOG=DEBUG to get more logging).

Just to let you know, this resource is using the old implementation, which we are moving away from. We will replace the implementation with the new SDK one soon, and we probably won't try to find the error in the current one.

**Comment by ayoul3** [comment]
Hey @sfc-gh-asawicki , 
Here is the specific statement that gives out an error

`2024-01-17T11:51:42.305+0100 [INFO]  provider.terraform-provider-snowflake_v0.68.1: 2024/01/17 11:51:42 [DEBUG] sql-conn-query [query DESCRIBE TABLE "RAW"."XXXXXXXX"."USERS" TYPE = COLUMNS; err <nil> duration 54.23575ms args {}]: timestamp=2024-01-17T11:51:42.305+0100
2024-01-17T11:51:42.306+0100 [INFO]  provider.terraform-provider-snowflake_v0.68.1: 2024/01/17 11:51:42 [DEBUG] sql-rows-next [err <nil> duration 12.458µs]: timestamp=2024-01-17T11:51:42.305+0100
2024-01-17T11:51:42.306+0100 [ERROR] provider.terraform-provider-snowflake_v0.68.1: Response contains error diagnostic: @caller=github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/diag/diagnostics.go:55 @module=sdk.proto diagnostic_summary="failed to parse result of describe: sql: expected 12 destination arguments in Scan, not 11" tf_req_id=b8f0cb73-f13e-9af2-3c87-a65c658d50e1 tf_resource_type=snowflake_table_column_masking_policy_application tf_rpc=ReadResource diagnostic_detail= diagnostic_severity=ERROR tf_proto_version=5.3 tf_provider_addr=registry.terraform.io/Snowflake-Labs/snowflake timestamp=2024-01-17T11:51:42.306+0100
`
No errors before. I wanna avoid posting the full debug log because it contains credentials. I can anonymize them however if this is not enough.

Thank you very much for your help

**Comment by sfc-gh-asawicki** [comment]
@ayoul3 thanks. It seems that the old implementation does not expect the last column (`privacy domain`) from here: https://docs.snowflake.com/en/sql-reference/sql/desc-table#examples (the last output).

As I mentioned, this old code should be migrated soon. We might address it earlier, though I will let you know.

**Comment by ayoul3** [comment]
thanks @sfc-gh-asawicki if it's possible to fix it, it would be very good because it's blocking everything (cannot destroy, cannot plan). I can remove the state and delete manually resources but I'd leave that as a last recourse.
In any case thank you for the extra speedy investigation ;)

**Comment by zarez** [comment]
Same here. Workaround is appreciated.  For now maybe just disable last night SF bundle

**Comment by mjbobak** [comment]
Can you apply the masking policy using snowflake_table resource? 
https://registry.terraform.io/providers/Snowflake-Labs/snowflake/0.76.0/docs/resources/table#masking_policy

**Comment by sfc-gh-swinkler** [comment]
@mjbobak you can but some people do not like to use the snowflake_table resource, as it doesn't handle renames of columns well at this time, so we have this other resource as an alternative instead. We do the same thing with object parameters which can also be set using the snowflake_object_parameter resource.

Agreed with @sfc-gh-asawicki , this is an old implementation, and we have a list of resources we need to refactor to use the new SDK, which will fix issues like these.

**Comment by sfc-gh-asawicki** [comment]
@ayoul3 I have merged the quick fix, it will be available in the nearest release (v0.84.0). We will probably roll it out today.

**Comment by ayoul3** [comment]
Thank you very much for the quick response !
